rezerwacje-app/src/main/java/org/niewidoczniakademicy/rezerwacje/service/HoursService.java

```diff
@@ -0,0 +1,88 @@
+package org.niewidoczniakademicy.rezerwacje.service;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.niewidoczniakademicy.rezerwacje.model.database.Hours;
+import org.niewidoczniakademicy.rezerwacje.model.database.Room;
+import org.niewidoczniakademicy.rezerwacje.model.rest.hours.AddHoursRequest;
+import org.niewidoczniakademicy.rezerwacje.model.rest.hours.TimeInterval;
+import org.niewidoczniakademicy.rezerwacje.model.shared.DayOfWeek;
+import org.niewidoczniakademicy.rezerwacje.service.converter.ConversionService;
+import org.niewidoczniakademicy.rezerwacje.service.exception.HoursOverlappingException;
+import org.niewidoczniakademicy.rezerwacje.service.exception.RoomNotFoundException;
+import org.niewidoczniakademicy.rezerwacje.service.repository.HoursRepository;
+import org.niewidoczniakademicy.rezerwacje.service.repository.RoomRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+@Slf4j
+@Service
+@AllArgsConstructor(onConstructor = @__(@Autowired))
+public final class HoursService {
+
+    private final RoomRepository roomRepository;
+    private final HoursRepository hoursRepository;
+    private final ConversionService conversionService;
+
+    public List<Hours> saveHours(final Map<DayOfWeek, List<TimeInterval>> availabilityDetails, final Long roomId) {
+        AddHoursRequest request = AddHoursRequest.builder()
+                .availabilityDetails(availabilityDetails)
+                .roomId(roomId)
+                .build();
+        return saveHours(request);
+    }
+
+    public List<Hours> saveHours(final AddHoursRequest request) {
+        final Room room = roomRepository.findById(request.getRoomId())
+                .orElseThrow(() -> new RoomNotFoundException("No room with id " + request.getRoomId()));
+
+        validateIfOverlapping(request, room.getAvailabilityHours());
```

Comment:
I think we should also validate if `timeStart < timeEnd` for all added TimeIntervals

Reply:
I forgot about that, added